src/main/java/com/blibli/future/controller/ApprovalController.java

```diff
@@ -0,0 +1,55 @@
+package com.blibli.future.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.blibli.future.service.api.ApprovalService;
+import com.blibli.future.vo.ApprovalRequestVo;
+import com.blibli.future.vo.ApprovalResponseVo;
+
+
+@RestController
+public class ApprovalController {
+	
+	public final String BASE_PATH = RequestController.BASE_PATH;
+	public final String PATH_APPR_OR_REJECT = RequestController.BASE_PATH + "/{type}/{id}";
+	
+	private ApprovalService approvalService;
+	
+	@Autowired
+	ApprovalController(ApprovalService approvalService){
+		this.approvalService = approvalService;
+	}
+	
+	@PutMapping(PATH_APPR_OR_REJECT)
+	public ResponseEntity<String> approving(@PathVariable String type , @PathVariable String id , @RequestBody ApprovalRequestVo approvalRequestVo){
+		if(type.equals("leave")){
+			approvalService.processLeave(id, approvalRequestVo.getNik(), approvalRequestVo.isApproved());
+		}
+		else if(type.equals("absence")){
+			approvalService.processAbsencePermit(id, approvalRequestVo.getNik(), approvalRequestVo.isApproved());
+		}
+		else{ return new ResponseEntity<String>("id: "+id+" not processed", HttpStatus.BAD_REQUEST); }
```

Comment:
Does this commit already has the `ControllerAdvice` class? If that's the case, just throw the exception and handle it there